### DISCUSSION TITLE
issue: Accessing `dirtyFields` first time from `formState` always gives empty object.

### DISCUSSION THREAD

**Discussion by nmanumr** [react-hook-form/react-hook-form#9447] [Q&A] [answered]
### Version Number

7.39.5

### Codesandbox/Expo snack

https://codesandbox.io/s/hook-form-y4t0tw

### Steps to reproduce

1. Go to the code sandbox link
2. Change any checkbox below and click "print dirty fields" it will be empty.
3. Now, change any other checkbox and click "print dirty fields" now it will give both the changed inputs.

### Expected behaviour

Checking the dirty fields for the first time should correctly output the dirty fields. 

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment, marked as answer]
> formState is wrapped with a [Proxy](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy) to improve render performance and skip extra logic if specific state is not subscribed to. Therefore make sure you invoke or read it before a render in order to enable the state update.

https://react-hook-form.com/api/useform/formstate

![Screenshot 2022-11-26 at 8 04 55 pm](https://user-images.githubusercontent.com/10513364/204081094-62900de0-fb82-4b6d-8eb9-758bc2e24f97.png)

**Comment by Phibee** [comment]
[
![Screen Shot 2022-11-28 at 3 04 49 PM](https://user-images.githubusercontent.com/3889714/204262994-99768783-d516-416d-bab4-861b1d52056f.png)
](url) 

i believe that's because when accessing formState does not include dirtyFields but once you destructure it you'll get the values. Like this formState: {dirtyFields} 

Hope it helps. 

Code:
https://codesandbox.io/s/hook-form-forked-xk2dw4?file=/src/App.js
